test-framework/junit5-vertx/src/main/java/io/quarkus/test/junit/vertx/UniAsserter.java

```diff
@@ -34,4 +34,18 @@
     UniAsserter assertTrue(Supplier<Uni<Boolean>> uni);
 
     UniAsserter assertFalse(Supplier<Uni<Boolean>> uni);
+
+    /**
+     * Used to determine whether or not the Uni contains the expected failure.
+     * The assertions fails if the Uni does not fail.
+     *
+     * @param c Consumer that performs custom assertions on whether or not the failure matches expectations.
```

Comment:
Maybe you could give an explicit hint about the possibility to run assertions on the exception message or cause from this consumer. Some Quarkus users might think it's not possible otherwise.

Reply:
Good idea. Done!